designsafe/static/scripts/workspace/services/apps-service.js

```diff
@@ -217,11 +276,11 @@ export function appsService($http, $q, $translate, djangoUrl, Django) {
         schema.properties.processorsPerNode = {
             title: 'Processors Per Node',
             description: `Number of processors (cores) per node for the job. e.g. A selection of 16 processors per node along with 4 nodes
-            will result in 16 processors on 4 nodes, with 64 processors total. Default number of processors per node is ${Math.floor(app.defaultProcessorsPerNode / app.defaultNodeCount)}.`,
+            will result in 4 nodes with 16 processors each, 64 processors total. Default number of processors per node is ${Math.floor(app.defaultProcessorsPerNode || 1) / (app.defaultNodeCount || 1)}.`,
             type: 'integer',
-            default: Math.floor(app.defaultProcessorsPerNode / app.defaultNodeCount),
+            default: Math.floor((app.defaultProcessorsPerNode || 1) / (app.defaultNodeCount || 1)),
             minimum: 1,
-            maximum: Math.floor(app.defaultProcessorsPerNode / app.defaultNodeCount),
+            maximum: Math.floor(app.defaultProcessorsPerNode || 1) / (app.defaultNodeCount || 1),
```

Comment:
Does `maximum` only need `app.defaultProcessorsPerNode` to be rounded or should it look like `default` too?

Reply:
It needs to look like default, since in the definition `defaultProcessorsPerNode` is actually a total of all processors across all nodes. So here we break it down into a value that actually represents `processorsPerNode`